src/test/java/org/prebid/server/bidder/readpeak/ReadPeakBidderTest.java

```diff
@@ -0,0 +1,443 @@
+        // given
+        final ObjectNode bidExt = mapper.createObjectNode()
+                .putPOJO("prebid", ExtBidPrebid.builder().type(banner).build());
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                BidRequest.builder()
+                        .imp(singletonList(Imp.builder().id("123").banner(Banner.builder().build()).build()))
+                        .build(),
+                mapper.writeValueAsString(
+                        givenBidResponse(bidBuilder -> bidBuilder.ext(bidExt).impid("123").mtype(2))));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isNotEmpty(); // This should pass
+        assertThat(result.getErrors())
+                .extracting(BidderError::getMessage)
+                .containsExactly("Unable to fetch mediaType 2 in multi-format: 123");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorForMissingMType() throws JsonProcessingException {
+        // given
+        final ObjectNode bidExt = mapper.createObjectNode()
+                .putPOJO("prebid", ExtBidPrebid.builder().type(banner).build());
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                BidRequest.builder()
+                        .imp(singletonList(Imp.builder().id("123").banner(Banner.builder().build()).build()))
+                        .build(),
+                mapper.writeValueAsString(
+                        givenBidResponse(bidBuilder -> bidBuilder.ext(bidExt).impid("123"))));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isNotEmpty();
+        assertThat(result.getErrors())
+                .extracting(BidderError::getMessage)
+                .containsExactly("Unable to fetch mediaType null in multi-format: 123");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldCreateBidExtIfMissing() throws JsonProcessingException {
+        // given
+        final Bid bid = Bid.builder()
+                .id("1")
+                .price(BigDecimal.valueOf(1.23))
+                .adm("${AUCTION_PRICE}")
+                .nurl("${AUCTION_PRICE}")
+                .impid("123")
+                .mtype(1)
+                .build();
+
+        final BidResponse bidResponse = BidResponse.builder()
+                .cur("USD")
+                .seatbid(List.of(SeatBid.builder()
+                        .bid(List.of(bid))
+                        .build()))
+                .build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(
+                BidRequest.builder()
+                        .imp(singletonList(Imp.builder().id("123").banner(Banner.builder().build()).build()))
+                        .build(),
```

Comment:
does a BidRequest has any impact on the result? just send null and hide it inside the `givenHttpCall`